code/__DEFINES/components.dm

```diff
@@ -11,12 +11,34 @@
 // All signals. Format:
 // When the signal is called: (signal arguments)
 
+// /datum signals
 #define COMSIG_COMPONENT_ADDED "component_added"				//when a component is added to a datum: (datum/component)
 #define COMSIG_COMPONENT_REMOVING "component_removing"			//before a component is removed from a datum because of RemoveComponent: (datum/component)
 #define COMSIG_PARENT_QDELETED "parent_qdeleted"				//before a datum's Destroy() is called: ()
-#define COMSIG_PARENT_ATTACKBY "atom_attackby"			        //from the base of atom/attackby: (obj/item, mob/living, params)
-#define COMSIG_PARENT_EXAMINE "atom_examine"                    //from the base of atom/examine: (mob)
+
+// /atom signals
+#define COMSIG_PARENT_ATTACKBY "atom_attackby"			        //from base of atom/attackby(): (obj/item, mob/living, params)
+#define COMSIG_ATOM_HULK_ATTACK "hulk_attack"					//from base of atom/attack_hulk(): (mob/living/carbon/human)
+#define COMSIG_PARENT_EXAMINE "atom_examine"                    //from base of atom/examine(): (mob)
 #define COMSIG_ATOM_ENTERED "atom_entered"                      //from base of atom/Entered(): (atom/movable, atom)
 #define COMSIG_ATOM_EX_ACT "atom_ex_act"						//from base of atom/ex_act(): (severity, target)
+#define COMSIG_ATOM_EMP_ACT "atom_emp_act"						//from base of atom/emp_act(): (severity)
+#define COMSIG_ATOM_FIRE_ACT "atom_fire_act"					//from base of atom/fire_act(): (exposed_temperature, exposed_volume)
+#define COMSIG_ATOM_BULLET_ACT "atom_bullet_act"				//from base of atom/bullet_act(): (obj/item/projectile, def_zone)

```

Comment:
holy copypaste batman

condense these into 1 atom_act and just check inside the signal for what type it is

Reply:
bad

Reply:
that seems like it would only hurt readability and performance checking the signal type when we already have a system for getting signals where they need to go

Reply:
Why would we hardcode each type

Reply:
@optimumtact